src/toil/jobStores/azureJobStore.py

```diff
@@ -784,26 +825,48 @@ def attempt(last=False):
         try:
             yield
         except retriable_exceptions as e:
-            # Any instance of these exceptions is automatically retriable.
-            if last:
-                raise
+            if time.time() < expiration:
```

Comment:
Please model this after https://github.com/BD2KGenomics/cgcloud/blob/9cfe6b894a7609244452e14e6f3ff90e0db8611d/lib/src/cgcloud/lib/ec2.py#L35